### PR TITLE
Limit scope of modeline search

### LINKED ISSUE #2942: C main.c is tagged as scheme
[This](https://github.com/BestImageViewer/geeqie/blob/1797b9e1a522b077f852aa560b35d0fca018c603/src/main.c) main.c that is clearly plain C is wrongly tagged as Scheme language.

Maybe that is linked to issue #2197.

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request limits the scope of the modeline search on files, as discussed in #2942.
It now only matches the first and last 5 lines against the modeline regular expressions.

Fixes #2942.

### CODE REVIEW

File: samples/C/main.c [hunk 1]
```diff
@@ -0,0 +1,931 @@
+/*
+ * Geeqie
+ * (C) 2006 John Ellis
+ * Copyright (C) 2008 - 2012 The Geeqie Team
+ *
+ * Author: John Ellis
+ *
+ * This software is released under the GNU General Public License (GNU GPL).
```
Comment (arfon) [inline samples/C/main.c]: Any chance you can find a non-GPL example @pchaigno?

Comment (Alhadis) [inline samples/C/main.c]: [Found one!](http://pastebin.com/tsd0ErxE) :D Or rather, edited one, after having cleaned up Google's crap code style.

GitHub: 1
GPL: 0

I AM THE THORN IN THE ARSE OF RICHARD STALLMAN.

Comment (pchaigno) [inline samples/C/main.c]: @Alhadis Thanks! I've updated the pull request with this sample.

**Comment by arfon** [comment]
:+1: thanks both.

**Pull request merged**